### ISSUE TITLE
GAN training unstable, strange losses.

### ISSUE THREAD

**Issue by lennert130994** [junyanz/pytorch-CycleGAN-and-pix2pix#702]
So i have a simple GAN with dense layers and leakyReLU activation, nothing special.
When the GAN is training correctly and the generated samples look good, the training loss looks like this:
![image](https://user-images.githubusercontent.com/23079356/61290125-666b0480-a7cb-11e9-8590-805b0654b6b1.png)
With blue being D and orange G.

A lot of times the GAN training goes wrong and the generated sample look like, well, shit. The training loss in this case looks like this:
![image](https://user-images.githubusercontent.com/23079356/61290134-6b2fb880-a7cb-11e9-869b-b5429e2e39e0.png)

Any help to what might cause this would be greatly appreciated!

**Comment by junyanz** [comment]
I don't know what has happened just based on the training loss curve. I recommend that you start with a good GAN [codebase](https://github.com/LMescheder/GAN_stability) and make changes gradually.
